### PR TITLE
test: tweak timeout and other details for lighthouse test

### PR DESCRIPTION
**Pull request by PolGuixe**
- Added a Jest test longer time-out default is too short for some cases and some slower machines
- Added conditional testing to ensure the test only runs when the server is available.

### CODE REVIEW

**Review by DSchau — CHANGES_REQUESTED** [review]
All of these changes look great; but the one note I would make is I'd remove the dependency on Bluebird. Why not just use the native Promise?

**Comment by PolGuixe** [comment]
@DSchau let me know what you think.

Next step would be to find a way to distinguish from local testing and CI/CD to run the test in different servers. 

Any ideas?

**Comment by DSchau** [comment]
also `package.json` (and correspondingly `yarn.lock`) should be updated with any new dependencies!

**Comment by PolGuixe** [comment]
@DSchau true... regarding the promise. 😉

**Comment by PolGuixe** [comment]
There were no new dependencies added sorry for the confusion.

`net` is part of the Node API

**Comment by DSchau** [comment]
@PolGuixe [`is-ci`](https://www.npmjs.com/package/is-ci) will likely be very helpful :)

File: lighthouse.test.js [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+const chromeLauncher = require('chrome-launcher')
 const Table = require('cli-table')
+const net = require('net')
 
-let table = new Table()
+const checkConnection = (host, port, timeout = 10000) => {
+  return new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      reject('timeout')
```
Comment (DSchau) [inline lighthouse.test.js]: Why not end the socket (if it exists) before rejecting?

File: lighthouse.test.js [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+      // eslint-disable-next-line no-use-before-define
+      socket.end()
+    }, timeout)
+    const socket = net.createConnection(port, host, () => {
+      clearTimeout(timer)
+      resolve(true)
+      socket.end()
```
Comment (DSchau) [inline lighthouse.test.js:14]: Inverse here.

Why not end the socket, and _then_ resolve?

Comment (DSchau) [reply lighthouse.test.js:14]: I'd also recommend `return resolve` or `return reject` because it tends to be clearer.

**Review by DSchau — COMMENTED** [review]
Left some more comments!

File: lighthouse.test.js (hunk 2 above)
Comment (PolGuixe) [reply lighthouse.test.js:14]: Same as above. 

I'll make another commit with the changes 😉

File: lighthouse.test.js (hunk 1 above)
Comment (PolGuixe) [reply lighthouse.test.js]: No special reason. You are right it looks clearer.

**Comment by PolGuixe** [comment]
On the last commit I also removed the eslint comments, maybe are not aplicable to this repo.

File: lighthouse.test.js [hunk 3]
```diff
@@ -12,26 +31,37 @@ const output = scores => {
...
+    }, {})
+
+    console.log(output(scores))
+
+    expect(scores.performance).toBe(1)
+    expect(scores.accessibility).toBe(1)
+    expect(scores['best-practices']).toBeGreaterThanOrEqual(0.93)
+    expect(scores.seo).toBe(1)
+  }
+  return
```
Comment (DSchau) [inline lighthouse.test.js]: Don’t need to return! Otherwise this looks great

**Review by DSchau — COMMENTED** [review]
Final comment!

**Review by DSchau — APPROVED** [review]
Looks great. Thanks for the PR!

File: lighthouse.test.js (hunk 3 above)
Comment (PolGuixe) [reply lighthouse.test.js]: done 😉

**Pull request merged**